[tanjieling0] iP

tanjieling0 opened the pull request:
# TommyBot
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes/))

TommyBot frees your mind for things that matter. Let us keep track of your tasks for you! It's
- text-based
- user friendly
- easy to learn
- ~FAST~ SUPER FAST to use

To get started with TommyBot:
1. Download `duke.jar` from [here](https://github.com/tanjieling0/ip/releases).
2. Locate the file on your desktop terminal
3. Run `java -jar duke.jar` (in the same folder as the jar file).
4. Let us do the managing of tasks for you 😆

And it is **FREE!**

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public static void main(String[] args) {
    new Duke("./data/tasks.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

tanjieling0 committed:
Rename, Greet, Exit

tanjieling0 committed:
Echo

tanjieling0 committed:
Add, List

tanjieling0 committed:
Mark as Done

tanjieling0 committed:
Add Task.java class

tanjieling0 committed:
Todo, Event, Deadline

tanjieling0 committed:
add ToString() to class

tanjieling0 committed:
Handle Errors

tanjieling0 committed:
Delete

tanjieling0 committed:
Automated Text UI Testing

tanjieling0 committed:
update input.txt

tanjieling0 committed:
Abstract divider printing

tanjieling0 committed:
Restructure codes

tanjieling0 committed:
Add Save function

tanjieling0 committed:
Merge branch 'branch-Level-7'

tanjieling0 committed:
Implement formatDate()

tanjieling0 committed:
Implement formatDate() for Deadline and Event

tanjieling0 committed:
Merge branch 'branch-Level-8'
Identification and conversion of dates to a uniform format

Dates are being stored as Strings which may not be as useful and
meaningful as LocalDate.

Identify and convert different date formats to a standard form of "MMM d
yyyy".

Allows us to input a variety of formats of dates while storing them in a
standard format. Makes it more standardised and cleaner.

tanjieling0 committed:
Implement OOP

tanjieling0 committed:
Add exceptions

tanjieling0 committed:
Add exceptions for Delete, Mark and Unmark

tanjieling0 committed:
Move formatDate() to Parser

tanjieling0 committed:
Merge branch 'A-MoreOOP'

tanjieling0 committed:
Tidy packages

tanjieling0 committed:
Merge branch 'A-Packages'

tanjieling0 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

tanjieling0 committed:
Set up Gradle

tanjieling0 committed:
Add Junit tests

tanjieling0 committed:
Merge branch 'A-JUnit'

tanjieling0 committed:
Adhere to coding standards

tanjieling0 committed:
Fix boolean function names

tanjieling0 committed:
Merge branch 'branch-A-CodingStandard'

tanjieling0 committed:
Add JavaDocs for commands

tanjieling0 committed:
Add JavaDocs for exceptions

tanjieling0 committed:
Add JavaDocs for tasks

tanjieling0 committed:
Add JavaDocs for Duke, Parser, Storage, Ui

tanjieling0 committed:
Resolve merge conflicts

tanjieling0 committed:
Fix OOP mistakes

tanjieling0 committed:
Implement find command

tanjieling0 committed:
Merge branch 'branch-Level-9'

tanjieling0 committed:
Add JavaDocs for thrown exceptions

tanjieling0 committed:
Refactor from Duke to Tommy

limcaaarl commented on src/main/java/tommy/Ui.java:
```diff
@@ -0,0 +1,177 @@
+ */
+public class Ui {
+
+    private static final String name = "Tommy";
```
Remember to follow the SCREAMING_SNAKE_CASE format for your constants!

```suggestion
    private static final String NAME= "Tommy";
```

limcaaarl commented on src/main/java/tommy/exception/TommyException.java:
```diff
@@ -0,0 +1,24 @@
+     *
+     * @return Descriptive error message.
+     */
+    public abstract String errorMessage();
```
Perhaps you could change the method name to a verb here.

```suggestion
    public abstract String getErrorMessage();
```

limcaaarl commented on src/main/java/tommy/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,48 @@
+    public void execute(Storage storage, TaskList taskList, Ui ui) throws InvalidArgumentException {
+        try {
+            String[] components = this.description.split(" /by ", 2);
+            String deadlineDetails = components[0];
```
Maybe you could change this to deadlineDetail instead to imply single-valued variable?

```suggestion
            String deadlineDetail = components[0];
```

limcaaarl commented on src/main/java/tommy/command/AddEventCommand.java:
```diff
@@ -0,0 +1,52 @@
+    public void execute(Storage storage, TaskList taskList, Ui ui) throws InvalidArgumentException {
+        try {
+            String[] components = this.description.split(" /from | /to ", 3);
+            String eventDetails = components[0];
```
Same for this one too!
```suggestion
            String eventDetail = components[0];
```

limcaaarl reviewed (commented):
I think this work is very high quality. The code is very easy to read and follow, especially with the proper naming used for the variables and methods. I made a few suggestions to some of the naming. But overall, you did a great job!👍👍👍

tanjieling0 committed:
Implement basic GUI

tanjieling0 committed:
Improve GUI aesthetics

tanjieling0 committed:
Merge branch 'branch-Level-10'

tanjieling0 committed:
Configure Checkstyle

tanjieling0 committed:
Merge checkstyle

tanjieling0 committed:
Fix greetings GUI

tanjieling0 committed:
Update JavaDocs

tanjieling0 committed:
Fix checkstyle

tanjieling0 committed:
Add assertions

tanjieling0 committed:
Improve code quality

There are multiple code quality violations within the code that makes
the code less readable. This includes magic values and multiple levels
of abstraction within the same code fragment.

Implementing Single Level of Abstraction Principle (SLAP) and replacing
magic values makes the code more understandable and readable for other
programmers. This also makes it easier to edit the code behaviour later
on since the values can be edited through the constants.

Let's extract all the magic values into constants.

Use SLAP where suitable, while leaving certain simple code fragments
that is already readable untouched.

tanjieling0 committed:
Merge pull request #2 from tanjieling0/branch-A-Assertions

Add assertions

tanjieling0 committed:
Handle merge conflicts

tanjieling0 committed:
Merge pull request #3 from tanjieling0/branch-A-CodeQuality

tanjieling0 committed:
Prevent duplicate tasks

Tasks can currently be duplicated within the task list, leading to
potential confusion and inefficiencies. Having duplicate tasks in the
task list can result in ambiguity.

Preventing duplicate tasks is more efficient as users do not have to
check if they have previously input the task.

Let's introduce validation checks during task addition to ensure that
duplicate tasks are not added to the list. Validating tasks before
addition ensures a cleaner task list and enhance user experience.
The proactive prevention of duplicates enhances the efficiency of
task management for the user.

tanjieling0 committed:
Merge pull request #4 from tanjieling0/branch-C-DetectDuplicates

Prevent duplicate tasks

tanjieling0 committed:
Fix text overrun issues

tanjieling0 committed:
Merge branch 'branch-A-BetterGui'

tanjieling0 committed:
Include user guide

tanjieling0 committed:
Fix user guide formatting
